### DISCUSSION TITLE
Proposal for CTRL-based IME Control Enhancement in Fcitx5

### DISCUSSION THREAD

**Discussion by TomiXRM** [fcitx/fcitx5#993] [Ideas]
Hello everyone,

I'm using Fcitx5 on Linux, and I have a suggestion regarding the current method of switching input methods. Inspired by the functionality provided by Karabiner-elements on MacOS, I would like to see a similar IME switching feature implemented in Fcitx5.

[Karabiner-elements | Toggle IME with using right-ctrl key and left-ctrl key](https://qiita.com/emonuh/items/ba5f12ef459f3a34f83e)

Specifically, I propose the ability to turn off the IME by pressing the right Ctrl key, and turn it on by pressing the left Ctrl key. This feature is widely used by MacOS users and would be particularly convenient for those using US keyboard layouts.

Furthermore, in my setup, the right Ctrl and left Ctrl keys are recognized as the same key, while the right Alt and left Alt keys function independently. Having Ctrl keys operate independently as well, similar to Alt keys, would provide greater flexibility in keyboard usage.

To the developers of Fcitx5, if feasible, I kindly request considering the implementation of this feature. I believe it would greatly enhance user convenience.

I look forward to hearing your thoughts and suggestions. Thank you very much.

**Comment by wengxt** [comment]
You should be able to do that if you setup the global config like this.

![image](https://github.com/fcitx/fcitx5/assets/259684/addbb94a-51bc-468a-a7d9-74178e12c4d0)
